crates/red_knot_python_semantic/src/types.rs

```diff
@@ -151,6 +165,73 @@ pub(crate) fn definitions_ty<'db>(
     }
 }
 
+/// Build a declared type from a [`DeclarationsIterator`].
+///
+/// Will return a union if there is more than one declaration, or at least one plus the possibility
+/// of undeclared.
+///
+/// If undeclared is a possibility, `Unknown` type will be part of the return type.
+///
+/// # Panics
+/// Will panic if there are no declarations and no possibility of undeclared. This is a logic
+/// error, as any symbol with zero live declarations clearly may (must!) be undeclared.
```

Comment:
What now, may or must?

```suggestion
/// error, as any symbol with zero live declarations clearly must be undeclared.
```